src/inmanta/data/__init__.py

```diff
@@ -4522,9 +4527,29 @@ async def get_resources(
         if not resource_version_ids:
             return []
         query_lock: str = lock.value if lock is not None else ""
-        query = f"SELECT * FROM {cls.table_name()} WHERE environment=$1 AND resource_version_id = ANY($2) {query_lock}"
+
+        def convert_or_ignore(rvid):
+            """Method to retain backward compatibility, ignore bad ID's"""
+            try:
+                return Id.parse_resource_version_id(rvid)
+            except Exception:
+                return None
+
+        parsed_rv = (convert_or_ignore(id) for id in resource_version_ids)
+        effective_parsed_rv = [id for id in parsed_rv if id is not None]
```

Comment:
Could you elaborate on why this is required/correct? Is it just for the unlikely case where we get invalid inputs we didn't reject before? If we merge this into iso6 only, can't we afford to break backwards compatibility?

Reply:
This is for backward compatibility, if you pass an invalid ID, it will simply be ignored.